### ISSUE TITLE
Bug: Reports "no-this-alias" with "Definition for rule . . . not found"

### ISSUE THREAD

**Issue by grapevinegizmos** [eslint/eslint#16640]
### Environment

Node version: 16.14.2
npm version:  8.5.0
Local ESLint version: 8.2.4.0 
Global ESLint version:  not found
Operating System: darwin 21.6.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
Node version: v16.14.2
npm version: v8.5.0
Local ESLint version: v8.24.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.6.0
```
</details>

```js
<!-- Paste your code here -->
rules: {
    "quotes": ["error", "double"],
    "import/no-unresolved": 0,
    "max-len": 0,
    "new-cap": 1,
    "require-jsdoc": 0,
    "valid-jsdoc": 0,
    "no-useless-escape": 1,
    "prefer-const": 0,
    "no-this-alias": 1,
    "no-prototype-builtins": 1,
  },

let o = this;
```

### What did you expect to happen?

I expected it would not report any error for the line `let o  = this`

### What actually happened?

    1:1    error    Definition for rule 'no-this-alias' was not found       no-this-alias
 
When I set "no-this-alias": 0, I get:
 
     60:11  error    Unexpected aliasing of 'this' to local variable  @typescript-eslint/no-this-alias

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

If I set "no-this-alias": 0 it reports as an error
let o = this;

**Comment by mdjermanovic** [comment]
Hi @grapevinegizmos, thanks for the issue!
 
> 1:1    error    Definition for rule 'no-this-alias' was not found       no-this-alias

This works as intended since there's no core rule `no-this-alias`. If you want to enable [`@typescript-eslint/no-this-alias`](https://typescript-eslint.io/rules/no-this-alias) plugin rule, then you need to add `@typescript-eslint/` prefix:

```diff
-"no-this-alias": 1,
+"@typescript-eslint/no-this-alias": 1,
